src/Controls/src/BindingSourceGen/BindingCodeWriter.cs

```diff
@@ -68,6 +46,28 @@ private static bool ShouldUseSetter(BindingMode mode, BindableProperty bindableP
 		//------------------------------------------------------------------------------
 		#nullable enable
 
+		namespace System.Runtime.CompilerServices
+		{
+			using System;
+			using System.CodeDom.Compiler;
+		
+			{{GeneratedCodeAttribute}}
+			[AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
+			file sealed class InterceptsLocationAttribute : Attribute
+			{
+				public InterceptsLocationAttribute(string filePath, int line, int column)
+				{
+					FilePath = filePath;
+					Line = line;
+					Column = column;
+				}
+		
+				public string FilePath { get; }
+				public int Line { get; }
+				public int Column { get; }
+			}
+		}
+
```

Comment:
This might seem like a weird change but we need to make the attribute file-scoped so that there aren't any collisions when there are assemblies with reference between them and they see each others internals (as in the case of `Microsoft.Maui.Controls.Core.dll` and `Microsoft.Maui.Controls.Compatibility.dll`